testplan/web_ui/testing/src/Report/BatchReportBeta/state/reportSlice.js

```diff
@@ -0,0 +1,65 @@
+// @ts-nocheck
+import _isObject from 'lodash/isObject';
+import { createSlice } from '@reduxjs/toolkit/dist/redux-toolkit.esm';
+import fetchReport from './reportActions/fetchReport';
+import Axios from 'axios';
+
+const DEFAULT_TIMEOUT = 60_000;
+
+export default createSlice({
+  name: 'report',
+  initialState: {
+    uid: null,
+    document: null,
+    isFetching: false,
+    isFetchCancelled: false,
+    fetchError: null,
+    fetchTimeout: DEFAULT_TIMEOUT,
+    lastResponseContentLength: null,
+  },
+  reducers: {
+    setReportUID: {
+      reducer(state, { payload }) { state.uid = payload; },
+      prepare: (uid = null) => ({ payload: uid }),
+    },
+    setFetchTimeout: {
+      reducer(state, { payload }) { state.fetchTimeout = payload; },
+      prepare: (timeout = DEFAULT_TIMEOUT) => ({ payload: timeout }),
+    },
+    setLastResponseContentLength: {
+      reducer(state, { payload }) {
+        state.lastResponseContentLength = payload;
+      },
+      prepare: (contentLength = 0) => ({
+        payload: typeof contentLength === 'string'
+          ? parseInt(contentLength)
+          : contentLength,
+      }),
+    },
+  },
+  extraReducers: {
+    [fetchReport.pending.type](state) {
+      state.isFetching = true;
+      state.isFetchCancelled = false;
+      state.fetchError = null;
+    },
+    [fetchReport.fulfilled.type](state, action) {
+      state.isFetching = false;
+      state.isFetchCancelled = false;
+      state.fetchError = null;
+      state.document = action.payload;
+    },
+    [fetchReport.rejected.type](state, action) {
+      state.isFetching = false;
+      if(_isObject(action.error) && action.error.message === 'Rejected') {
+        // handled with rejectWithValue
+        const { payload: rejectValue } = action;
+        state.isFetchCancelled = Axios.isCancel(rejectValue);
+        state.fetchError = rejectValue;
+      } else {
+        state.isFetchCancelled = false;
+        state.fetchError = action.error;
+      }
```

Comment:
It seems fetchReport has 3 different state: pending/fulfilled/rejected, and we are mapping it 2 flags: isFetching/isFetchCancelled + a fetchError field. Can we simplify things here?